ravenframework/SupervisedLearning/SupervisedLearning.py

```diff
@@ -335,9 +335,9 @@ def evaluate(self,edict):
     names, values  = list(edict.keys()), list(edict.values())
     stepInFeatures = 0
     for index in range(len(values)):
-      #If value is a float, convert to numpy array for evaluation
-      if type(values[index]).__name__ == 'float':
-        values[index] = np.array(values[index])
+      # If value is a float or int, convert to numpy array for evaluation
+      if isinstance(values[index], (float, int)):
```

Comment:
you should use:

https://github.com/idaholab/raven/blob/a733699db3fb1d9e9e0fa7fb3377d18488326753/ravenframework/utils/mathUtils.py#L752

Reply:
Hm, isAFloatOrInt also is true with np.number so possibly the current code is good.

Reply:
Line 340 is needed because `np.array(values[index])` would create an array with shape `()`. Downstream of this RAVEN attempts to access the array with an index and fails because there is no index. I think it's weird that numpy creates an array with no shape instead of `(1,)`, but maybe they have their reasons.

Reply:
Okay, I am fine with this.

Reply:
```np.number``` is the generic class for any scalar in numpy so if for whatever reason the value is casted as (for example) ```np.int32```, this check would still work. 
Just my unrequested 2 cents ;)